src/psyclone/domain/nemo/transformations/nemo_arrayrange2loop_trans.py

```diff
@@ -308,16 +302,71 @@ def validate(self, node, options=None):
                 "supplied node argument should be within an ArrayReference "
                 "node that is within the left-hand-side of an Assignment "
                 "node, but it is on the right-hand-side.")
-        # Does the rhs of the assignment have any operations that
-        # return arrays as this is not currently supported?
+        # Does the rhs of the assignment have any operations that are not
+        # elemental?
         for operation in assignment.rhs.walk(Operation):
-            # At the moment the only array valued operator is matmul
-            if operation.operator == BinaryOperation.Operator.MATMUL:
+            # Allow non elemental UBOUND and LBOUND
+            if operation.operator is BinaryOperation.Operator.LBOUND:
+                continue
+            if operation.operator is BinaryOperation.Operator.UBOUND:
+                continue
+            if not operation.is_elemental():
                 raise TransformationError(
                     "Error in NemoArrayRange2LoopTrans transformation. This "
                     "transformation does not support array valued operations "
```

Comment:
Please could you  update this error message as I don't think it's quite right at the moment. e.g. `MAX` is elemental and so, given arrays as arguments, returns an array.

Reply:
Done